src/prelaunch/berzap.a

```diff
@@ -24,20 +25,67 @@ cheat1
          jmp   $b000
 
 cheat2
-         +DISABLE_ACCEL
-         lda   $b2cf
-         cmp   #$2b
-         bne   +
-         +GET_MACHINE_STATUS
+         +GET_MACHINE_STATUS_AND_MOCKINGBOARDSLOT
          and   #CHEATS_ENABLED
          beq   +
          lda   #$ad
          sta   $3d94      ; patch - don't decrease lives
          sta   $6573      ; patch - don't increase lives
          sta   $76ba      ; patch - don't increase lives
-+
+
++        txa   
+         beq   skipmb
+         stx   $3FCD      ;Replace #$C4 with detected slot
+         stx   $3FD5      ;Page $3F is relocated to $1F
+         stx   $3FE1
+         stx   $3FE6
+         stx   $3FEB
+         stx   $3FF0
+         stx   $AB1D
+firstPage
+         ldy   #$32       ;Start scan at first patch $9743
+-        lda   $9711,Y    ;Find #$C4's in code (24 locations) 
+         cmp   #$C4
+         bne   +
+         txa
+         sta   $9711,Y    ;Replace #$C4 with detected slot
++        iny              ;Finish scan at $9811
+         bne   -
+secondPage
+-        lda   $9811,Y    ;Find #$C4's in code (17 locations) 
+         cmp   #$C4
+         bne   +
+         txa
+         sta   $9811,Y    ;Replace #$C4 with detected slot
+         iny              ;Skip next 2 bytes after a match because there is never another match so close
+         iny
++        iny              ;Finish scan at $9911 = (last patch 990E)+3
+                          ;Check math: 9911-9743=1CE code length; 00 index end - CE code length = 32 index start
+         bne   -
+
+thirdPage
+         ldy   #$7B       ;Start scan at first patch $ACE0
+-        lda   $AC65,Y    ;Find #$C4's in code (21 locations) 
+         cmp   #$C4
+         bne   +
+         txa
+         sta   $AC65,Y    ;Replace #$C4 with detected slot
+         iny              ;Skip next 2 bytes after a match because there is never another match so close
+         iny
++        iny              ;Finish scan at $AD65 = (last patch AD62)+3
+                          ;Check math: AD65-ACE0=85 code length; 00 index end - 85 code length = 7B  index start
+         bne   -
+
+skipmb
+         lda   #$6C       ;Remove patch from code so it is only run once
+         sta   $B20B
+         lda   #$CE
+         sta   $B20C
+         lda   #$B2
+         sta   $B20D
+         +DISABLE_ACCEL
          jmp   ($b2ce)
 
-!if * > $1C0 {
+!if * > $1F0 {
```

Comment:
Yeah, we can't do this. `$1C0` is a hard limit outside of our control. Something about SCSI devices I think. qkumba would know for sure. Anyway, we need to find a way to get all this code to fit below $1C0. qkumba, any ideas?

Reply:
IIc+ uses over 32 bytes so that's $1E0, and SCSI uses even more, so I chose $1C0 to be safe.
The prelaunch code has access to the file system - it can load a second-stage prelaunch to another memory page if we get desperate, but we might be able to offer a routine in LC that accepts offset/value pairs in a table.  That should fit in the stack page.  I'll look into it.

Reply:
Left it in for now.